### ISSUE TITLE
twisted.web2.client cuts non-chunked responses of indeterminate length short

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3136]
|<img alt="andrew_w's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| andrew_w reported|
|-|-|
|Trac ID|trac#3136|
|Type|defect|
|Created|2008-03-29 00:14:49Z|

twisted.web2.client cuts responses short if they are not in chunked encoding and do not have a Content-Length header. I have attached a patch that fixes this bug.

Attachments:

* [twisted-web2-client-close-as-end.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/205/205e123cd0ad6c76344e88c1f36c22cd1528fa67/ad5c38bc5277c543ba6cc1c27fbef62bdb2b3334.patch) (3186 bytes) - added by andrew_w on 2008-03-29 00:15:54Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3136 3136
type__defect defect
reporter__andrew_w andrew_w
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__web2 web2
keywords__ 
time__1206749689000000 1206749689000000
changetime__1297616873000000 1297616873000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set status to `closed`|
|-|-|

Not going to fix this in web2 (See [#4821](https://github.com/twisted/twisted/issues/4821)).  The new Twisted Web client API (`twisted.web.client.Agent`) knows how to deal with responses that have no content-length header, so not retargeting this at the web component.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
